### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'uri'

### ISSUE THREAD

**Issue by Notaghost9997** [DevLARLEY/wingfox-downloader#1]
python .\main.py --id 174125 --cookie wingfox.com_cookies.txt
seed 22
hlsLevel web
hlsPrivate 1
Traceback (most recent call last):
  File "C:\github\wingfox-downloader\main.py", line 314, in <module>
    modified_data = dl.implant_token(
                    ^^^^^^^^^^^^^^^^^
  File "C:\github\wingfox-downloader\main.py", line 184, in implant_token
    uri=key.uri,
        ^^^^^^^
AttributeError: 'NoneType' object has no attribute 'uri'

m3u8 response:
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-ALLOW-CACHE:YES
#EXT-X-KEY:METHOD=AES-128,URI="https://hls.videocc.net/9215d65496/0/9215d6549696c666a3b25993db41cc60_1.key",IV=0xa1410f00a64ed2a95b6ed12cab5c8449
#EXT-X-TARGETDURATION:11
#EXTINF:10.042667,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_0.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_1.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_2.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_3.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_4.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_5.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_6.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_7.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_8.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_9.pts
#EXTINF:9.700000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_10.pts
#EXT-X-ENDLIST

also had to change from
f'OWtjN9xcDcc2cwXKxECpRgKw7piD4RwCdfOUlyNHFdSV0gHi={body.get('seed_const')}'
to 
f"OWtjN9xcDcc2cwXKxECpRgKw7piD4RwCdfOUlyNHFdSV0gHi={body.get('seed_const')}"

**Comment by DevLARLEY** [comment]
You don't even need a token. I'm just running: `python main.py --id 174125 --cookie some` and it works...

**Comment by Notaghost9997** [comment]
https://github.com/user-attachments/assets/204a52b6-2c81-435c-922c-72da28fdde54

**Comment by DevLARLEY** [comment]
I can tell you that the cookie shouldn't be a file but rather the cookie itself but that doesn't matter in this case.
Could you print 'm3u8_data' at line 176 in main.py?
![image](https://github.com/user-attachments/assets/83186d3e-6e4e-4af2-8360-3f5184ef2f24)

**Comment by Notaghost9997** [comment]
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-ALLOW-CACHE:YES
#EXT-X-KEY:METHOD=AES-128,URI="https://hls.videocc.net/9215d65496/0/9215d6549696c666a3b25993db41cc60_1.key",IV=0xa1410f00a64ed2a95b6ed12cab5c8449
#EXT-X-TARGETDURATION:11
#EXTINF:10.042667,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_0.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_1.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_2.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_3.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_4.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_5.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_6.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_7.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_8.pts
#EXTINF:10.000000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_9.pts
#EXTINF:9.700000,
https://hw-dts.videocc.net/9215d65496/0/1234567890123/0/41/cc/60_1/9215d6549696c666a3b25993db41cc60_1_10.pts
#EXT-X-ENDLIST
```
please use this for cookies
```
from http.cookiejar import MozillaCookieJar
def get_cookies(cookiefile):
        cj = MozillaCookieJar(cookiefile, policy=False)
        cj.load(ignore_discard=True, ignore_expires=True) # 
        cookies = dict()
        for cookie in cj:
            cookies[cookie.name] = cookie.value
        return cookies 
```

**Comment by DevLARLEY** [comment]
Did you install the requirements.txt file using Python 3.12?

**Comment by Notaghost9997** [comment]
they are already installed 
```
~\Videos\github\wingfox-downloader main ❯ pip install -r .\requirements.txt
Requirement already satisfied: m3u8 in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from -r .\requirements.txt (line 1)) (3.4.0)
Requirement already satisfied: pycryptodome in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from -r .\requirements.txt (line 2)) (3.20.0)
Requirement already satisfied: requests in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from -r .\requirements.txt (line 3)) (2.32.3)
Requirement already satisfied: iso8601 in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from m3u8->-r .\requirements.txt (line 1)) (1.1.0)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from requests->-r .\requirements.txt (line 3)) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from requests->-r .\requirements.txt (line 3)) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from requests->-r .\requirements.txt (line 3)) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\x\appdata\local\programs\python\python311\lib\site-packages (from requests->-r .\requirements.txt (line 3)) (2024.6.2)

[notice] A new release of pip is available: 23.1.2 -> 24.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip 
```

**Comment by DevLARLEY** [comment]
I don't know why it's not working
![image](https://github.com/user-attachments/assets/8d2b6c13-b030-4950-9ddf-8f466499f03a)
I guess try using python3.12?
